src/pyPROPOSAL/detail/secondaries.cxx

```diff
@@ -16,52 +18,114 @@
-    py::class_<secondaries::NaivIonization, secondaries::Ionization, std::shared_ptr<secondaries::NaivIonization>>(m_sub, "NaivIonization");
-
-    py::class_<secondaries::MupairProduction, secondaries::Parametrization, std::shared_ptr<secondaries::MupairProduction>>(m_sub, "MupairProduction")
-        .def("calculate_rho", &secondaries::MupairProduction::CalculateRho,
-            py::arg("energy"), py::arg("v_loss"), py::arg("component"), py::arg("rnd1"), py::arg("rnd2"));
-    py::class_<secondaries::KelnerKokoulinPetrukhinMupairProduction, secondaries::MupairProduction, std::shared_ptr<secondaries::KelnerKokoulinPetrukhinMupairProduction>>(m_sub, "KelnerKokoulinPetrukhinMupairProduction");
-
-    py::class_<secondaries::Photonuclear, secondaries::Parametrization, std::shared_ptr<secondaries::Photonuclear>>(m_sub, "Photonuclear");
-
-    py::class_<secondaries::PhotopairProduction, secondaries::Parametrization, std::shared_ptr<secondaries::PhotopairProduction>>(m_sub, "PhotopairProduction");
-    py::class_<secondaries::PhotoTsai, secondaries::PhotopairProduction, std::shared_ptr<secondaries::PhotoTsai>>(m_sub, "PhotoTsai");
-
-    py::class_<secondaries::WeakInteraction, secondaries::Parametrization, std::shared_ptr<secondaries::WeakInteraction>>(m_sub, "WeakInteraction");
-    py::class_<secondaries::NaivWeakInteraction, secondaries::WeakInteraction, std::shared_ptr<secondaries::NaivWeakInteraction>>(m_sub, "NaivWeakInteraction");
-
-
-    py::class_<SecondariesCalculator, std::shared_ptr<SecondariesCalculator>>(m_sub, "SecondariesCalculator")
-        .def(py::init<std::vector<InteractionType>, ParticleDef const&, Medium const&>())
+    py::class_<secondaries::Parametrization,
+        std::shared_ptr<secondaries::Parametrization>>(m_sub, "Parametrization")
+        .def_property_readonly("random_numbers",
+            &secondaries::Parametrization::RequiredRandomNumbers)
+        .def("interaction_type",
+            &secondaries::Parametrization::GetInteractionType)
+        .def("calculate_secondaries",
+            &secondaries::Parametrization::CalculateSecondaries);
+
+    SecondariesBuilder<secondaries::Annihilation, secondaries::Parametrization> {}
+        .decl_virtuell_param(m_sub, "Annihilation");
+    SecondariesBuilder<secondaries::Bremsstrahlung,
+        secondaries::Parametrization> {}
+        .decl_virtuell_param(m_sub, "Bremsstrahlung");
+    SecondariesBuilder<secondaries::Compton, secondaries::Parametrization> {}
+        .decl_virtuell_param(m_sub, "Compton");
+    SecondariesBuilder<secondaries::EpairProduction,
+        secondaries::Parametrization> {}
+        .decl_virtuell_param(m_sub, "EpairProduction");
+    SecondariesBuilder<secondaries::Ionization, secondaries::Parametrization> {}
+        .decl_virtuell_param(m_sub, "Ionization");
+    SecondariesBuilder<secondaries::MupairProduction,
+        secondaries::Parametrization> {}
+        .decl_virtuell_param(m_sub, "MupairProduction");
```

Comment:
Shouldn't this also be `decl_rho_param`?

Reply:
yes

Reply:
Oh, no, it does not, my mistake :D